File: Data Structures & Algorithms/is-anagram/submission-1.py
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        map = {}

        for c in s:
            map[c] = map.get(c, 0) + 1
        
        for c in t:
            if c not in map or map[c] == 0:
                return False
            else:
                map[c] = map[c] - 1
        
        for val in map.values():
            if val > 0:
                return False
        return True
            

# class Solution:
#     def isAnagram(self, s: str, t: str) -> bool:
#         map = {}

#         for c in s:
#             map[c] = map.get(c, 0) + 1
        
#         for c in t:
#             if c not in map or map[c] == 0:
#                 return False
#             map[c] = map[c] - 1
        
#         for val in map.values():
#             if val > 0:
#                 return False
#         return True

File: Data Structures & Algorithms/valid-sudoku/submission-1.cpp
class Solution {
public:
    bool isValidSudoku(vector<vector<char>>& board) {
        
        unordered_set<string> seen;

        int row = board.size();
        int col = board[0].size();

        for (int r = 0; r < row; r++) {
            for (int c = 0; c < col; c++) {
                
                if (board[r][c] == '.') {
                    continue;
                }

                string patternR = "r" + to_string(r) + board[r][c];
                string patternC = "c" + to_string(c) + board[r][c];
                string patternB = "B" + to_string(r / 3) + to_string(c / 3) + board[r][c];

                if (seen.find(patternR) != seen.end() ||
                    seen.find(patternC) != seen.end() ||
                    seen.find(patternB) != seen.end()) {
                    return false;
                }

                seen.insert(patternR);
                seen.insert(patternC);
                seen.insert(patternB);
            }
        }

        return true;
    }
};